File: ea/combination.py
def helper(arr, s, i, path):
    if s == len(path):
        print(sorted(path))
        return 

    n = len(arr)
    for j in range(i, n):
        helper(arr, s, j, path+[arr[j]])


def combine(arr, s):
  helper(arr, s, 0, ['F', 'M', 'D'])


combine(['F', 'M', 'D'], 4)

File: ea/combination.js
const _ = require('lodash');

function generateCombinations(sidePosition) {
    const helper = (substring, path) => {
        if (path) res.add(path);
        if (!substring) return;

        for (const j of _.range(substring.length)) {
            helper(substring.slice(j+1), path+substring[j]);
        }
    };

    const res = new Set();
    helper(sidePosition, '');
    return res;
}

console.log(generateCombinations('DFFM'));